cloudant-http/src/main/java/com/cloudant/http/HttpConnection.java

```diff
@@ -354,6 +355,25 @@ public HttpConnection execute() throws IOException {
             // retry flag is set from the final step in the response interceptRequest pipeline
             retry = currentContext.replayRequest;
 
+            // If we're going to retry we should consume any existing error streams to avoid
+            // leaking connections. Consuming the stream is preferable to just closing it as it
+            // makes the connection eligible for re-use.
+            if (retry && numberOfRetries > 0) {
+                InputStream errorStream = connection.getErrorStream();
+                if (errorStream != null) {
+                    try {
+                        // Copy the stream to a null destination
+                        IOUtils.copy(errorStream, NullOutputStream.NULL_OUTPUT_STREAM);
+                        logger.log(Level.WARNING,"Consumed unused HTTP response error stream.");
```

Comment:
Will the stream be set to `null` if it has been already consumed? Is there a chance that it can still be non-null but return 0 bytes from the copy?

Reply:
It is only `null` if there was no error or the "server provided no useful information". Generally the stream will have been closed if it was consumed, but there is a chance it could return 0 bytes from the copy.

Reply:
So should the warning log entry only be made if `IOUtils.copy` returns non-0?

Reply:
Also done in the utility method I added in 0687f06.